osu.Game/Screens/Edit/Timing/EffectSection.cs

```diff
@@ -41,6 +41,12 @@ protected override void LoadComplete()
             omitBarLine.Current.BindValueChanged(_ => saveChanges());
             scrollSpeedSlider.Current.BindValueChanged(_ => saveChanges());
 
+            // adjusting scroll speed on osu/catch rulesets results in undefined behaviour during legacy beatmap decoding, and generally shouldn't be shown.
+            // todo: there should be proper way to identify such rulesets, but this should do for now.
```

Comment:
I mean... you could manufacture one right now already? There's no reason why you wouldn't be able to just do something like

```csharp
var drawableRulesetTemporary = ruleset.CreateDrawableRulesetWith(Beatmap);
if (drawableRulesetTemporary is IDrawableScrollingRuleset scrollingRuleset
        && scrollingRuleset.VisualisationMethod != ScrollVisualisationMethod.Constant)
{
    // ...
}
```

`IDrawableScrollingRuleset` does not exist right now, but it's trivially extractable. It's maybe a touch wasteful, but otherwise this sort of way seems like a really low bar effort-wise for making this a lot less hacky.

Reply:
I thought about going with this approach initially, but I wasn't sure about rulesets that provide custom implementation which use that field. Probably won't hurt to have anything non-hacky at least.